### ISSUE TITLE
lbs dies on APPEND with an incorrect starting block number

### ISSUE THREAD

**Issue by gperciva** [Tarsnap/kivaloo#282]
If we send an APPEND to an lbs server with an incorrect starting block number, the server does an `exit(1)`.  However,  `INTERFACES` says:

```
APPEND: Request type = 0x00000002
...
        Response if the starting block # is incorrect:
        [4 byte status code = 1]
```

so that sounds like the server should be able to recover.

However, `storage_write()` contains:
```
        /* Sanity-check the write position. */
        if (blkno != S->nextblk) {
                warn0("Attempt to append data with wrong blkno");
                warn0("(%016" PRIx64 ", should be %016" PRIx64 ")",
                    blkno, S->nextblk);
                goto err2;
        }
```

which goes back to `worker.c`'s `workthread`:
```
                case 1: /* Write */
                        if (storage_write(ctl->sstate,
                            ctl->blkno, ctl->nblks, ctl->buf)) {
                                warnp("Failure writing blocks");
                                exit(1);
                        }
                        break;
```

I'm happy to either:
- modify `INTERFACES` to eliminate this response, or
- modify the lbs server to recover from an APPEND with a bad starting block number

**Comment by cperciva** [comment]
Hmm, we should validate things like this (and send back an error response) at the dispatch level.  If the storage-layer code is checking it should be just as a belts-and-suspenders thing.

**Comment by gperciva** [comment]
Fixed in #283 .
